main.go

```diff
@@ -110,5 +115,23 @@ func main() {
 		log.Fatal(err)
 	}
 
+	if opts.ProfilerOutput != "" {
+		f, err := os.Create(opts.ProfilerOutput)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if err := pprof.StartCPUProfile(f); err != nil {
```

Comment:
We should use net/http/pprof instead. I think there is a code in exporter_shared for it; if not, please add it to that shared package

Reply:
:+1: